src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/convert/ObjectMapperFactory.java

```diff
@@ -6,22 +6,37 @@
         
 package com.microsoft.azure.spring.data.cosmosdb.core.convert;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.module.SimpleModule;
 import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
 import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
 
+import java.time.ZonedDateTime;
+
 public class ObjectMapperFactory {
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
     static {
+        OBJECT_MAPPER.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         OBJECT_MAPPER.registerModule(new ParameterNamesModule())
                         .registerModule(new Jdk8Module())
                         .registerModule(new JavaTimeModule());
+
+        if (!OBJECT_MAPPER.canDeserialize(OBJECT_MAPPER.constructType(ZonedDateTime.class))) {
```

Comment:
Is this just a safety catch or a convention when adding modules ?

Reply:
The JavaTimeModule should covert this type. I'll remove it.